build.rs

```diff
@@ -11,11 +11,15 @@ fn main() {
 
     match env::var("CARGO_CFG_TARGET_OS").unwrap().as_ref() {
         "linux" => {
-            // statically link libunwind if compiling for musl, dynamically link otherwise
+            // statically link libunwind if statically linking musl (e.g. the
+            // `crt-static` target feature is present), dynamically link
+            // otherwise
             if env::var("CARGO_FEATURE_UNWIND").is_ok() {
                 println!("cargo:rustc-cfg=use_libunwind");
                 if env::var("CARGO_CFG_TARGET_ENV").unwrap() == "musl"
-                    && env::var("CARGO_CFG_TARGET_VENDOR").unwrap() != "alpine"
+                    && !env::var("CARGO_CFG_TARGET_FEATURE")
```

Comment:
This logic is backwards. If `CARGO_CFG_TARGET_FEATURE` contains `crt-static` then we want to statically link.
```suggestion
                    && env::var("CARGO_CFG_TARGET_FEATURE")
```